website/src/content/blog/2025-03-31-typespec-1-0-release/type_spec_1_0_release.md

```diff
@@ -0,0 +1,160 @@
+
+> **TypeSpec-generated SDKs ensure seamless integration with your backend, since everything is derived from the same single source of truth.**
+
+For example, a generated TypeScript SDK might allow you to interact with your `todo` API like this:
+
+```typescript
+const client = new TodoServiceClient();
+const todo = await client.createTodoItem({
+  content: "Buy groceries",
+  dueDate: "2025-03-21",
+  isCompleted: false,
+});
+console.log(todo);
+```
+
+### Key benefits:
+
+- **Accelerated development**: Jumpstart your server and client implementations with automatically generated models, controllers, and SDKs.
+- **API consistency**: Ensure your server and client implementations adhere precisely to your API specification.
+- **Reduced manual effort**: Eliminate repetitive coding tasks, freeing you to focus on business logic.
+- **Improved maintainability**: Easily update your server and client code as your API evolves by regenerating the stubs and SDKs from the updated TypeSpec definition.
+
+## User Journey: Building a "To Do" Service with TypeSpec
+
+Imagine building a "to do" service API. With TypeSpec, you start by defining your data models and operations in a concise and human-readable format. For example:
+
+```typescript
+@route("/todoitems")
+namespace TodoItems {
+  @get
+  op getTodoItems(): GetTodoItemsResponse;
+
+  model GetTodoItemsResponse {
+    @statusCode statusCode: 200;
+    @body todoitems: TodoItem[];
+  }
+
+  @post
+  op createTodoItem(@body body: CreateTodoItemRequest): CreateTodoItemResponse;
+
+  model CreateTodoItemResponse {
+    @statusCode statusCode: 201;
+    @body todoitem: TodoItem;
+  }
+}
+
+model TodoItem {
+  id: string;
+  content: string;
+  isCompleted: boolean;
+  labels: string[];
+}
+
+model CreateTodoItemRequest {
+  content: string;
+  labels?: string[];
+}
+```
+
+From this definition, TypeSpec generates the foundational server-side code, including controllers and models. Here’s an example of a generated controller method:
+
+```csharp
+[HttpGet]
+[Route("/todoitems")]
+[ProducesResponseType((int)HttpStatusCode.OK, Type = typeof(TodoItem[]))]
+public virtual async Task<IActionResult> GetTodoItems()
+{
+    var result = await TodoItemsOperationsImpl.GetTodoItemsAsync();
+    return Ok(result);
+}
+```
+
+And a corresponding model:
+
+```csharp
+public partial class TodoItem
+{
+    public string Id { get; set; }
+    public string Content { get; set; }
+    public bool IsCompleted { get; set; }
+    public string[] Labels { get; set; }
+}
+```
+
+With the generated code in place, you can quickly integrate it into your project, build the service, and run it. Testing the API is just as straightforward. For instance, you can use a tool like Thunder Client to send a POST request:
+
+```json
+{
+  "content": "Buy groceries",
+  "labels": ["shopping", "urgent"]
+}
+```
+
+TypeSpec ensures that your API implementation stays perfectly aligned with your specification, saving you time and effort.
```

Comment:
It is great showcasing the TypeSpec client/service code gen. Should we have step by step tutorial (linked by separate from this blog post) that folks can follow along and get hands-on?

Reply:
Dina is working on that.

Reply:
on second thought, the content Dina is working on won't be available till Build. I'll see what I can put together this week.